### PR TITLE
Don't allow randomizer to select invisible maps

### PR DESCRIPTION
**Pull request by LittleEndu**
Before, randomizer might select beatmaps from other rulesets which made the carousel not scroll to anything. This makes it so that randomizer can only select from visible beatmaps of the selected set.

### CODE REVIEW

File: osu.Game/Screens/Select/BeatmapCarousel.cs [hunk 1]
```diff
@@ -249,20 +249,21 @@ public void SelectNextRandom()
 
             if (RandomAlgorithm == RandomSelectAlgorithm.RandomPermutation)
             {
-                var notYetVisitedSets = visible.Except(previouslyVisitedRandomSets).ToList();
+                var notYetVisitedSets = visibleSets.Except(previouslyVisitedRandomSets).ToList();
                 if (!notYetVisitedSets.Any())
                 {
-                    previouslyVisitedRandomSets.Clear();
-                    notYetVisitedSets = visible;
+                    previouslyVisitedRandomSets.RemoveAll(s => visibleSets.Contains(s));
```
Comment (smoogipoo) [inline osu.Game/Screens/Select/BeatmapCarousel.cs]: Does this really make a functional difference than just `Clear()`ing?

**Review by smoogipoo — COMMENTED** [review]
This needs a test case in TestCaseBeatmapCarousel.

**Comment by peppy** [comment]
Would like to see a testcase added for this one!

File: osu.Game/Screens/Select/BeatmapCarousel.cs (hunk 1 above)
Comment (LittleEndu) [reply osu.Game/Screens/Select/BeatmapCarousel.cs]: Let's say you randomize through half of your maps then you wanna play only maps from some artist so you add a filter. Now you run out of maps while the filter is still on. Right now the code would also reset that 50% of other maps you were playing before even though you have more maps the randomizer hasn't selected yet if you didn't have a filter on. 

Just thought some users might experience that and then say that the randomizer is still not random enough.

**Review by peppy — APPROVED** [review]

**Pull request merged**